### DISCUSSION TITLE
How to enable inlay hint with clangd

### DISCUSSION THREAD

**Discussion by nixmd** [joaotavora/eglot#1196] [Q&A] [answered]
I've read that recent versions of eglot supports inlay-hints for clangd. I've tried multiple attempts explained [here](https://emacs.stackexchange.com/questions/76473/how-to-enable-inlay-hint-in-eglot-clangd) to enable it with no success. It would be great if there was some quick notes on how to enable this feature.

**Comment by joaotavora** [comment, marked as answer]
You could start by saying which `clangd` version you are using.  I've tested with `clangd` 15 and upwards, both on Linux and Windows.

You don't need any of the configuration you post, and your interpretation in that linked question is also incorrect.  

> It would be great if there was some quick notes on how to enable this feature.

This LSP feature is enabled by default in Eglot, so there are no notes to make about that.

If you have a recent enough `clangd` in `PATH` and have a file `thingy.cpp` containing this silly program:
```
int foo(int bar){return bar*bar;}
auto baz = foo(42);
```

Then simply `emacs -Q thingy.cpp -f eglot` should eventually show you two inlay hints on the second line.

**Reply by nixmd** [reply]
```
$ clangd --version
Ubuntu clangd version 14.0.0-1ubuntu1
Features: linux+grpc
Platform: x86_64-pc-linux-gnu
```
```
$ emacs --version
GNU Emacs 30.0.50
Development version 3748c8a62e80 on master branch; build date 2023-03-19.
```
eglot is the one which is brought by emacs and 1.13.

Just tried `emacs -Q thingy.cpp -f eglot` with no success.
To make sure that if this version of clangd supports inlay-hints or not, installed and checked vscodium and it did, but it does not work with the setup I mention above.

**Reply by nixmd** [reply]
Just added llvm repositories to my distro and installed clangd-15, fixed symlink to clangd and restarted eglot server and it works 😁
Thanks.

**Reply by joaotavora** [reply]
OK, good.  It could work maybe even with version 14, I don't know.  If you want to check, pull up `M-x eglot-events-buffer` and looks at the server capabilities.
